src/components/BookFinder/BookFinder.tsx

```diff
@@ -1,41 +1,48 @@
-import React, { FunctionComponent, useEffect, useState } from "react";
+import React, {
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useState,
+} from "react";
 import { BookSearch } from "./BookSearch/BookSearch";
 import { BookList } from "./BookList/BookList";
 import axios from "axios";
 import { notification } from "antd";
+import { searchUrl } from "../../helpers/BookHelper";
 
 export const BookFinder: FunctionComponent = () => {
   const [query, setQuery] = useState("");
   const [books, setBooks] = useState([]);
   const [total, setTotal] = useState(0);
   const [isLoading, setIsLoading] = useState(false);
   const [startIndex, setStartIndex] = useState(0);
-  const url = "https://www.googleapis.com/books/v1/volumes?q=";
 
-  useEffect(() => {
-    const fetchBooks = async () => {
-      setIsLoading(true);
+  const fetchBooks = useCallback(async () => {
+    setIsLoading(true);
 
-      try {
-        const result = await axios(`${url}${query}&startIndex=${startIndex}`);
+    try {
+      const result = await axios(
+        `${searchUrl}${query}&startIndex=${startIndex}`
+      );
 
-        setIsLoading(false);
-        setTotal(result.data.totalItems);
-        setBooks(result.data.items);
-      } catch (e) {
-        notification.error({
-          message: "An error occurred",
-          description: "Please try searching again",
-        });
-      }
+      setIsLoading(false);
+      setTotal(result.data.totalItems);
+      setBooks(result.data.items);
+    } catch (e) {
+      notification.error({
+        message: "An error occurred",
+        description: "Please try searching again",
+      });
+    }
 
-      window.scrollTo(0, 0);
-    };
+    window.scrollTo(0, 0);
+  }, [query, startIndex]);
```

Comment:
I think you should also put searchUrl here as a dependency as well. PHPStorm should give you a warning if incorrect dependencies are present or if some are missing (it shows at the opening bracket). Not 100% sure on this one though, it won't impact much (it's not like the searchUrl is going to change a whole lot).

Reply:
iz nekog razloga mi ne javlja tu error za search url, al mi ima smisla da bi trebal pa dodam :D